frontend/src/hooks/useFileExport.ts

```diff
@@ -0,0 +1,84 @@
+import axios from "axios";
+import { useSnackbar } from "notistack";
+import { useCallback } from "react";
+import { useSelector } from "react-redux";
+import { selectDocument } from "../store/documentSlice";
+import { selectEditor } from "../store/editorSlice";
+
+export const enum FileExtension {
+	Markdown = "markdown",
+	HTML = "html",
+	PDF = "pdf",
+}
+
+interface UseFileExportReturn {
+	handleExportToPDF: () => void;
+	handleExportToHTML: () => void;
+	handleExportToMarkdown: () => void;
+}
+
+export const useFileExport = (): UseFileExportReturn => {
+	const { enqueueSnackbar } = useSnackbar();
+	const editorStore = useSelector(selectEditor);
+	const documentStore = useSelector(selectDocument);
+
+	const markdown = editorStore.doc?.getRoot().content?.toString() || "";
+	const documentName = documentStore.data?.title || "codepair_document";
```

Comment:
I think that we can reduce computational cost by moving into `handleExportFile`.